app/components/pilas-blockly.js

```diff
@@ -414,19 +414,30 @@ export default Component.extend({
   },
 
   showBlocksErrorExpectationFeedback(){
+    this.showExpectationFeedbackFor(
+      warningInControlStructureBlock,
+      addWarning,
+      getNestedControlStructureBlocks
+    )
     this.showExpectationFeedbackFor(
       isCritical,
       addError
     )
   },
 
-  showExpectationFeedbackFor(condition, addFeedback) {
+  declarationWithNameToArray(declaration){
+    return [declarationWithName(declaration)]
+  },
+
+  showExpectationFeedbackFor(condition, addFeedback, getBlocks = this.declarationWithNameToArray) {
```

Comment:
Creeeeo que es más expresivo sin nombrarlo, y además más corto.....

```suggestion
  showExpectationFeedbackFor(condition, addFeedback, getBlocks = (n) => [declarationWithName(n)]) {
```